common/app/templates/inlineJS/nonBlocking/detectAdblock.scala.js

```diff
@@ -0,0 +1,29 @@
+try {
+    ((document, window) => {
+        window.guardian.adBlockers = {};
+        var ad = document.createElement('div');
+        ad.style.position = 'absolute';
+        ad.style.left = '-9999px';
+        ad.style.height = '10px';
+        ad.innerHTML = '&nbsp;';
+        ad.setAttribute('class', 'ad_unit');
+
+        // avoid a forced sync layout, and open door to more accurate detection
+        // e.g. detecting network behaviour
+        window.requestAnimationFrame(() => {
+            document.body.appendChild(ad);
+
+            // avoid a forced layout, and be sure the element has been added to the DOM
+            window.requestAnimationFrame(() => {
+                var adStyles = window.getComputedStyle(ad);
+                window.guardian.adBlockers.generic = adStyles.getPropertyValue('display') === 'none';
+                window.guardian.adBlockers.ffAdblockPlus = adStyles.getPropertyValue('-moz-binding').match('elemhidehit') !== null;
```

Comment:
I know this is merged but I think this is causing `Cannot read property 'match' of null` in the console ( I guess because `getPropertyValue('-moz-binding')` only has a return value in firefox?). Would put in a fix, but didn't want to mess about too much with something that seemed fairly fragile from the conversations on this PR!